libc/string_lenchr.c

```diff
@@ -150,9 +154,291 @@ TEST(string_len, big)
+
+	/* Testing capability of functions to hold and read not ascii set */
+	sz = sizeof(notAsciiSet);
+	TEST_ASSERT_EQUAL_STRING(notAsciiSet, strchr(notAsciiSet, notAsciiSet[0]));
+	TEST_ASSERT_EQUAL_STRING(notAsciiSet, strrchr(notAsciiSet, notAsciiSet[0]));
+	TEST_ASSERT_EQUAL_STRING(notAsciiSet, memchr(notAsciiSet, notAsciiSet[0], sz));
+
+	TEST_ASSERT_EQUAL_STRING(&notAsciiSet[64], strchr(notAsciiSet, notAsciiSet[64]));
+	TEST_ASSERT_EQUAL_STRING(&notAsciiSet[64], strrchr(notAsciiSet, notAsciiSet[64]));
+	TEST_ASSERT_EQUAL_STRING(&notAsciiSet[64], memchr(notAsciiSet, notAsciiSet[64], sz));
+
+	TEST_ASSERT_EQUAL_STRING(&notAsciiSet[sz - 1], strchr(notAsciiSet, notAsciiSet[sz - 1]));
+	TEST_ASSERT_EQUAL_STRING(&notAsciiSet[sz - 1], strrchr(notAsciiSet, notAsciiSet[sz - 1]));
+	TEST_ASSERT_EQUAL_STRING(&notAsciiSet[sz - 1], memchr(notAsciiSet, notAsciiSet[sz - 1], sz));
+}
+
+
+TEST(string_chr, torn)
+{
+	char *torn = "foo\0bar";
+	int sz;
+
+	/* Checking correct working of null terminating point */
+	sz = strlen(torn) + 1;
+	TEST_ASSERT_EQUAL_STRING("", strchr(torn, '\0'));
+	TEST_ASSERT_EQUAL_STRING("", strrchr(torn, '\0'));
+	TEST_ASSERT_EQUAL_STRING("", memchr(torn, '\0', sz));
+
+	TEST_ASSERT_EQUAL_STRING("foo", strchr(torn, 'f'));
+	TEST_ASSERT_EQUAL_STRING("foo", strrchr(torn, 'f'));
+	TEST_ASSERT_EQUAL_STRING("foo", memchr(torn, 'f', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(torn, 'b'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(torn, 'b'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(torn, 'b', sz));
+}
+
+
+TEST(string_chr, whitespaces)
+{
+	const char whites[] = " \v\t\r\n";
+	int sz;
+
+	/* Checking if white signs doesn't make any interference on functions output */
+	sz = sizeof(whites) - 1;
+	TEST_ASSERT_EQUAL_STRING(&whites[sz - 1], strchr(whites, whites[sz - 1]));
+	TEST_ASSERT_EQUAL_STRING(&whites[sz - 1], strrchr(whites, whites[sz - 1]));
+	TEST_ASSERT_EQUAL_STRING(&whites[sz - 1], memchr(whites, whites[sz - 1], sz));
+
+	TEST_ASSERT_EQUAL_STRING(&whites[sz / 2], strchr(whites, whites[sz / 2]));
+	TEST_ASSERT_EQUAL_STRING(&whites[sz / 2], strrchr(whites, whites[sz / 2]));
+	TEST_ASSERT_EQUAL_STRING(&whites[sz / 2], memchr(whites, whites[sz / 2], sz));
+
+	TEST_ASSERT_EQUAL_STRING(&whites[0], strchr(whites, whites[0]));
+	TEST_ASSERT_EQUAL_STRING(&whites[0], strrchr(whites, whites[0]));
+	TEST_ASSERT_EQUAL_STRING(&whites[0], memchr(whites, whites[0], sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(whites, 'x', sz));
+}
+
+
+TEST(string_chr, empty)
+{
+	int sz;
+
+	/* Checking if we can get an empty string */
+	sz = 1;
+	TEST_ASSERT_EQUAL_STRING("", strchr("", '\0'));
+	TEST_ASSERT_EQUAL_STRING("", strrchr("", '\0'));
+	TEST_ASSERT_EQUAL_STRING("", memchr("", '\0', sz));
+
+	TEST_ASSERT_EQUAL_STRING("", strchr("", 0));
+	TEST_ASSERT_EQUAL_STRING("", strrchr("", 0));
+	TEST_ASSERT_EQUAL_STRING("", memchr("", 0, sz));
+
```

Comment:
\0 and 0 are the same character

```suggestion
	TEST_ASSERT_EQUAL_STRING("", strchr("", '\0'));
	TEST_ASSERT_EQUAL_STRING("", strrchr("", '\0'));
	TEST_ASSERT_EQUAL_STRING("", memchr("", '\0', sz));
```